core/src/avm2/property.rs

```diff
@@ -64,56 +65,24 @@ impl<'gc> PropertyClass<'gc> {
             PropertyClass::Class(class) => (Some(*class), false),
             PropertyClass::Name(gc) => {
                 let (name, unit) = &**gc;
-                let outcome = resolve_class_private(name, *unit, activation)?;
-                let class = match outcome {
-                    ResolveOutcome::Class(class) => {
+                if name.is_any_name() {
+                    *self = PropertyClass::Any;
+                    (None, true)
+                } else {
+                    // Note - we look up the class in the domain by name, which allows us to look up private classes.
+                    // This also has the advantage of letting us coerce to a class while the `ClassObject`
+                    // is still being constructed (since the `Class` will already exist in the domain)
+                    let domain = unit.map_or(activation.avm2().stage_domain, |u| u.domain());
```

Comment:
correct me if I'm wrong: I think `unit` can only be `None` if this comes from a builtin Rust-defined trait, so I think this should use a builtin domain instead?

Reply:
You're right - I've changed it.